src/wxProfilerGUI/database.cpp

```diff
@@ -255,27 +294,53 @@ void Database::loadCallstacks(wxInputStream &file,bool collapseKernelCalls)
 
 	while (!file.Eof())
 	{
-		wxString line = str.ReadLine();
-		if (line.IsEmpty())
+		wxString lineAddr = str.ReadLine();
+		wxString lineSamples = str.ReadLine();
+		if (lineAddr.IsEmpty() || lineSamples.IsEmpty())
 			break;
 
-		std::wistringstream stream(line.c_str().AsWChar());
+		std::wistringstream streamAddr(lineAddr.c_str().AsWChar());
 
 		CallStack callstack;
-		stream >> callstack.samplecount;
+		AddrInfo *topAddrInfo = NULL;
 
+		int depth = 0;
 		while (true)
 		{
 			std::wstring addrstr;
-			stream >> addrstr;
+			streamAddr >> addrstr;
 			if (addrstr.empty())
 				break;
 			Address addr = hexStringTo64UInt(addrstr);
 
+			if (depth == 0)
+			{
+				topAddrInfo = &addrinfo.at(addr);
+			}
+

```

Comment:
So I guess this is `IPCounts.txt`'s replacement.

Reply:
That's right. Since the sample count is no longer a single number per address, I figured it's better to reconstruct it at load time rather than record/write/read it twice.